src/gadgets/k_mix.rs

```diff
@@ -1,61 +1,213 @@
 #![allow(non_snake_case)]
 
 use super::mix;
-use bulletproofs::r1cs::ConstraintSystem;
-use error::SpacesuitError;
+use bulletproofs::r1cs::{ConstraintSystem, R1CSError};
+use curve25519_dalek::scalar::Scalar;
 use std::iter::once;
```

Comment:
Nitpick: for top-level funcs it's often easier to read the code if their immediate namespace is spelled out. Because you can do that, both Rust and Go prefer to have short non-elaborate names for functions, which in turn reinforces the desire to spell out the namespace in front of the function call.

What i mean is, import may look like `use std::iter` and then the code would do `iter::once(...)` instead of `once(...)`.